pocs/dome/protocol_astrohaven_simulator.py

```diff
@@ -96,48 +96,58 @@ def __init__(self, command_queue, status_queue, stop):
         self.command_queue = command_queue
         self.status_queue = status_queue
         self.stop = stop
+        self.logger = logger
         self.delta = datetime.timedelta(seconds=1)
         self.shutter_a = Shutter('A', Protocol.OPEN_A, Protocol.CLOSE_A, Protocol.A_OPEN_LIMIT,
-                                 Protocol.A_CLOSE_LIMIT)
+                                 Protocol.A_CLOSE_LIMIT, self.logger)
         self.shutter_b = Shutter('B', Protocol.OPEN_B, Protocol.CLOSE_B, Protocol.B_OPEN_LIMIT,
-                                 Protocol.B_CLOSE_LIMIT)
+                                 Protocol.B_CLOSE_LIMIT, self.logger)
         self.next_output_code = None
         self.next_output_time = None
+        self.logger.info('AstrohavenPLCSimulator created')
+
+    def __del__(self):
+        if not self.stop.is_set():
+            self.logger.critical('AstrohavenPLCSimulator.__del__ stop is NOT set')
 
     def run(self):
+        self.logger.info('AstrohavenPLCSimulator.run ENTER')
         self.next_output_time = datetime.datetime.now()
         while True:
             if self.stop.is_set():
-                # print('Returning from AstrohavenPLCSimulator.run', file=sys.stderr)
+                self.logger.info('Returning from AstrohavenPLCSimulator.run EXIT')
                 return
             now = datetime.datetime.now()
             remaining = (self.next_output_time - now).total_seconds()
-            # print('AstrohavenPLCSimulator.run remaining=%r' % remaining, file=sys.stderr)
+            self.logger.info('AstrohavenPLCSimulator.run remaining=%r' % remaining)
             if remaining <= 0:
                 self.do_output()
-                self.update_next_output_time()
                 continue
             # Maybe a short delay here?
```

Comment:
This comment was here before but is it referring to same thing as short delay added on 132?

Reply:
Doh, should have fixed that. Will do so now.